bricks/template/__brick__/{{project_name.snakeCase()}}/lib/usecases/user/get_users_use_case.dart

```diff
@@ -0,0 +1,23 @@
+import 'package:{{project_name.snakeCase()}}/api/repository/credential_repository.dart';
+import 'package:{{project_name.snakeCase()}}/usecases/base/base_use_case.dart';
+import 'package:{{project_name.snakeCase()}}/model/user.dart';
+import 'package:injectable/injectable.dart';
+
+@Injectable()
+class GetUsersUseCase extends NoParamsUseCase<List<User>> {
+  final CredentialRepository _credentialRepository;
+
+  GetUsersUseCase(this._credentialRepository);
+
+  @override
+  Future<Result<List<User>>> call() async {
+    try {
+      final userResponses = await _credentialRepository.getUsers();
+      final users =
+          userResponses.map((userResponse) => userResponse.toUser()).toList();
+      return Success(users);
+    } catch (exception) {
+      return Failed(UseCaseException(exception));
+    }
```

Comment:
Can we use the Future's operators here instead of a try/catch? We caught the exceptions in the Repository layer already 🤔  

For example: https://github.com/luongvo/flutter-survey/blob/8de8375734c323d507aa89ce777d97db3ed87a21/lib/usecase/get_user_profile_use_case.dart#L13-L21

Reply:
@luongvo I'm afraid that we cannot use `Future operators` here. At the Repository, we catch the Network error only. Meanwhile, we catch inside the UseCase for all (error from network or mapping, for example) 🙏

Reply:
@doannimble I don't think so. According to the implementation here in the Repository layer, 

https://github.com/nimblehq/flutter-templates/blob/ab5559ccd8c79dd819f1f538303d4a4f54b92148/bricks/template/__brick__/%7B%7Bproject_name.snakeCase()%7D%7D/lib/api/repository/credential_repository.dart#L14-L21

we caught all the possible exceptions and wrapped them with NetworkExceptions already (please check the `else` block from the method below `fromDioException`). 

https://github.com/nimblehq/flutter-templates/blob/ab5559ccd8c79dd819f1f538303d4a4f54b92148/sample/lib/api/exception/network_exceptions.dart#L46-L128

I can not get what's the new error from mapping here. 🤔 
So, using `.then` to get the Success value and `.onError` for the Failed value is possible 💭 

![image](https://github.com/nimblehq/flutter-templates/assets/16315358/17e22d96-3858-4f90-ba03-c1a4a2431ca7)